modcc/CMakeLists.txt

```diff
@@ -30,17 +30,32 @@ set(libmodcc_sources
 set(modcc_sources modcc.cpp)
 
 add_library(libmodcc STATIC ${libmodcc_sources})
-target_include_directories(libmodcc PUBLIC .)
+target_link_libraries(libmodcc PUBLIC arbor-public-headers)
```

Comment:
As far as I understand, to export the symbols of the various libraries we need to build arbor as a shared library using the CMake option `-DBUILD_SHARED_LIBS=ON`. I tried this and noticed that `arbor`, `arborio`, `arborenv` and `arbor-sup` all get built as shared libraries, but `libmodcc` is built as a static library. I'm far from a CMake pro, if you could point out where that gets decided that would be really helpful.

Reply:
that's because I forced it to be static here https://github.com/boeschf/arbor/blob/f38aece0d9bb1d8a14e92022e3354083ef80447e/modcc/CMakeLists.txt#L32
There were some problems building it dynamically (I have to check what happens exactly). The library gets used to make the `modcc` compiler (only?) so I thought it would be ok to leave it static.

Reply:
`add_library` without `STATIC` or `SHARED` are compiled as static library by default unless you set `-DBUILD_SHARED_LIBS=ON`